lms/static/js/commerce/views/receipt_view.js

```diff
@@ -53,7 +54,7 @@ var edx = edx || {};
             if (orderId && this.$el.data('is-payment-complete')==='True') {
                 // Get the order details
                 self.$el.removeClass('hidden');
-                self.getReceiptData(orderId).then(self.renderReceipt, self.renderError);
+                self.getReceiptData(orderId).then(self.getProviderData, self.renderError);
```

Comment:
Two concerns here:

1) Does this mean that we can't render *any* of the receipt until we get the provider data?  It seems like we should be able to render at least the section that's common to all receipts.

2) It's really hard to follow this code, since the receipt rendering doesn't happen until three levels of callbacks have occurred.

I'd prefer if this were structured like:
1) The receipt page structure renders immediately (I think this is already the case)
2) In parallel:
    a) Retrieve the order information and render it in the receipt page.
    b) If the order contains a credit seat, retrieve the provider information and render it in the receipt page.